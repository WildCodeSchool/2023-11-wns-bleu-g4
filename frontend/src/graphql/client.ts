import { ApolloClient, InMemoryCache } from "@apollo/client";
const uri = process.env.NEXT_PUBLIC_GRAPHQL_API_URL;

export const cache = new InMemoryCache();

const client = new ApolloClient({
  uri: uri || "/graphql",
  cache: cache,
  credentials: "include",
});

export default client;