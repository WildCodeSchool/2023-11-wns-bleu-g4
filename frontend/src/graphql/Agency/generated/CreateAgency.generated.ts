import * as Types from '../../generated/schema';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type CreateAgencyMutationVariables = Types.Exact<{
  data: Types.NewAgencyInput;
}>;


export type CreateAgencyMutation = { __typename?: 'Mutation', createAgency: { __typename?: 'Agency', id: number, name: string, address: string, postcode: string, city: string, country: string, phone: string, email: string, bookings: Array<{ __typename?: 'Booking', id: number }>, productCodes: Array<{ __typename?: 'ProductCode', id: number }> } };


export const CreateAgencyDocument = gql`
    mutation CreateAgency($data: NewAgencyInput!) {
  createAgency(data: $data) {
    id
    name
    address
    postcode
    city
    country
    phone
    email
    bookings {
      id
    }
    productCodes {
      id
    }
  }
}
    `;
export type CreateAgencyMutationFn = Apollo.MutationFunction<CreateAgencyMutation, CreateAgencyMutationVariables>;

/**
 * __useCreateAgencyMutation__
 *
 * To run a mutation, you first call `useCreateAgencyMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useCreateAgencyMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [createAgencyMutation, { data, loading, error }] = useCreateAgencyMutation({
 *   variables: {
 *      data: // value for 'data'
 *   },
 * });
 */
export function useCreateAgencyMutation(baseOptions?: Apollo.MutationHookOptions<CreateAgencyMutation, CreateAgencyMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<CreateAgencyMutation, CreateAgencyMutationVariables>(CreateAgencyDocument, options);
      }
export type CreateAgencyMutationHookResult = ReturnType<typeof useCreateAgencyMutation>;
export type CreateAgencyMutationResult = Apollo.MutationResult<CreateAgencyMutation>;
export type CreateAgencyMutationOptions = Apollo.BaseMutationOptions<CreateAgencyMutation, CreateAgencyMutationVariables>;