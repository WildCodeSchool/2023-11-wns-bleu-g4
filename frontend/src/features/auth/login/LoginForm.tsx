import { ToastConfigLogin } from "@/config/ToastConfig";
import { useLoginMutation } from "@/graphql/User/generated/Login.generated";
import { useProfileQuery } from "@/graphql/User/generated/Profile.generated";
import {
  Box,
  Button,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  Divider,
  Flex,
  FormControl,
  FormLabel,
  Heading,
  Input,
  InputGroup,
  InputRightElement,
  LightMode,
  Text,
} from "@chakra-ui/react";
import { useTranslation } from "next-i18next";
import Link from "next/link";
import React, { FormEvent } from "react";
import { toast } from "react-toastify";

export default function LoginForm() {
  const { t } = useTranslation("LoginForm");

  const [login] = useLoginMutation();
  const [show, setShow] = React.useState(false);
  const handleClick = () => setShow(!show);

  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    const formData = new FormData(e.target as HTMLFormElement);
    const formJSON: any = Object.fromEntries(formData.entries());
    // const profile = useProfileQuery()

    try {
      await login({ variables: { data: formJSON } });
      toast.info("LOGIN SUCCESSFULL", ToastConfigLogin);
      window.location.replace("/");
    } catch (e: any) {
      const errArr = e.message.replace("_", " ");
      toast.error(errArr, ToastConfigLogin);
      return;
    }


  };

  return (
    <Card variant="loginCard" boxShadow="md" w={{ base: "300px", sm: "396px" }} h="fit-content">
      {/* TITLE */}
      <CardHeader textAlign="center">
        <Heading as="h1" color="black" fontWeight="500">
          {t("LOGIN")}
        </Heading>
      </CardHeader>
      <Divider color="black" />
      <form onSubmit={handleSubmit}>
        <CardBody>
          <Flex direction="column" gap="15px">
            {/* EMAIL */}
            <FormControl size="md">
              <FormLabel>Email</FormLabel>
              <LightMode>
                <InputGroup>
                  <Input
                    type="email"
                    color="black"
                    placeholder="Email"
                    name="email"
                    size="md"
                    bg="bgLight"
                    borderRadius="lg"
                    required
                  />
                </InputGroup>
              </LightMode>
            </FormControl>

            {/* PASSWORD */}
            <Box>
              <FormControl size="md">
                <FormLabel>{t("Password")}</FormLabel>
                <LightMode>
                  <InputGroup>
                    <Input
                      type={show ? "text" : "password"}
                      color="black"
                      placeholder={t("Password")}
                      name="password"
                      size="md"
                      bg="bgLight"
                      borderRadius="lg"
                      required
                    />
                    <InputRightElement width="4.5rem">
                      <Button h="1.75rem" size="sm" onClick={handleClick}>
                        {show ? t("Hide") : t("Show")}
                      </Button>
                    </InputRightElement>
                  </InputGroup>
                </LightMode>
              </FormControl>

              {/* FORGOT PASSWORD */}
              <Text className=" text-center text-sm py-2">
                <Link href="#" className="hover:underline hover:text-orange-500">
                  {t("Forgot your password ?")}
                </Link>
              </Text>
            </Box>
          </Flex>
        </CardBody>
        <CardFooter>
          <Flex direction="column" className="w-full">
            {/* BUTTON */}
            <Button type="submit" className="w-full" variant="loginButton" m="0">
              {t("Login")}
            </Button>
            {/* FORGOT PASSWORD */}
            <Text className=" text-center text-sm py-2" color="black">
              {t("Not yet registered ?")}&nbsp;
              <Link href="/signup" className="underline text-orange-500">
                {t("sign up")}
              </Link>
            </Text>
          </Flex>
        </CardFooter>
      </form>
    </Card>
  );
}