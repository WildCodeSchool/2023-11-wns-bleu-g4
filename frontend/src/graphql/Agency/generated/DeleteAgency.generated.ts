import * as Types from "../../generated/schema";

import { gql } from "@apollo/client";
import * as Apollo from "@apollo/client";
const defaultOptions = {} as const;
export type DeleteAgencyMutationVariables = Types.Exact<{
  agencyId: Types.Scalars["Float"]["input"];
}>;

export type DeleteAgencyMutation = { __typename?: "Mutation"; deleteAgency: string };

export const DeleteAgencyDocument = gql`
  mutation DeleteAgency($agencyId: Float!) {
    deleteAgency(agencyId: $agencyId)
  }
`;
export type DeleteAgencyMutationFn = Apollo.MutationFunction<DeleteAgencyMutation, DeleteAgencyMutationVariables>;

/**
 * __useDeleteAgencyMutation__
 *
 * To run a mutation, you first call `useDeleteAgencyMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useDeleteAgencyMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [deleteAgencyMutation, { data, loading, error }] = useDeleteAgencyMutation({
 *   variables: {
 *      agencyId: // value for 'agencyId'
 *   },
 * });
 */
export function useDeleteAgencyMutation(
  baseOptions?: Apollo.MutationHookOptions<DeleteAgencyMutation, DeleteAgencyMutationVariables>,
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useMutation<DeleteAgencyMutation, DeleteAgencyMutationVariables>(DeleteAgencyDocument, options);
}
export type DeleteAgencyMutationHookResult = ReturnType<typeof useDeleteAgencyMutation>;
export type DeleteAgencyMutationResult = Apollo.MutationResult<DeleteAgencyMutation>;
export type DeleteAgencyMutationOptions = Apollo.BaseMutationOptions<
  DeleteAgencyMutation,
  DeleteAgencyMutationVariables
>;
