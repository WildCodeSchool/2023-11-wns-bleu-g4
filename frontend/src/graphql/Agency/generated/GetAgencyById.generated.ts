import * as Types from '../../generated/schema';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type GetAgencyByIdQueryVariables = Types.Exact<{
  agencyId: Types.Scalars['Int']['input'];
}>;


export type GetAgencyByIdQuery = { __typename?: 'Query', getAgencyById: { __typename?: 'Agency', address: string, city: string, country: string, email: string, id: number, name: string, phone: string, postcode: string, productCodes: Array<{ __typename?: 'ProductCode', id: number, size?: string | null, bookingItems: Array<{ __typename?: 'BookingItem', endDate: any, startDate: any }>, product?: { __typename?: 'Product', id: number } | null }> } };


export const GetAgencyByIdDocument = gql`
    query GetAgencyById($agencyId: Int!) {
  getAgencyById(agencyId: $agencyId) {
    address
    city
    country
    email
    id
    name
    phone
    postcode
    productCodes {
      id
      bookingItems {
        endDate
        startDate
      }
      size
      product {
        id
      }
    }
  }
}
    `;

/**
 * __useGetAgencyByIdQuery__
 *
 * To run a query within a React component, call `useGetAgencyByIdQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetAgencyByIdQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetAgencyByIdQuery({
 *   variables: {
 *      agencyId: // value for 'agencyId'
 *   },
 * });
 */
export function useGetAgencyByIdQuery(baseOptions: Apollo.QueryHookOptions<GetAgencyByIdQuery, GetAgencyByIdQueryVariables> & ({ variables: GetAgencyByIdQueryVariables; skip?: boolean; } | { skip: boolean; }) ) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<GetAgencyByIdQuery, GetAgencyByIdQueryVariables>(GetAgencyByIdDocument, options);
      }
export function useGetAgencyByIdLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<GetAgencyByIdQuery, GetAgencyByIdQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<GetAgencyByIdQuery, GetAgencyByIdQueryVariables>(GetAgencyByIdDocument, options);
        }
export function useGetAgencyByIdSuspenseQuery(baseOptions?: Apollo.SuspenseQueryHookOptions<GetAgencyByIdQuery, GetAgencyByIdQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useSuspenseQuery<GetAgencyByIdQuery, GetAgencyByIdQueryVariables>(GetAgencyByIdDocument, options);
        }
export type GetAgencyByIdQueryHookResult = ReturnType<typeof useGetAgencyByIdQuery>;
export type GetAgencyByIdLazyQueryHookResult = ReturnType<typeof useGetAgencyByIdLazyQuery>;
export type GetAgencyByIdSuspenseQueryHookResult = ReturnType<typeof useGetAgencyByIdSuspenseQuery>;
export type GetAgencyByIdQueryResult = Apollo.QueryResult<GetAgencyByIdQuery, GetAgencyByIdQueryVariables>;