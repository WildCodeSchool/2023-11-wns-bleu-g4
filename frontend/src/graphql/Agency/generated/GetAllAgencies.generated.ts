import * as Types from "../../generated/schema";

import { gql } from "@apollo/client";
import * as Apollo from "@apollo/client";
const defaultOptions = {} as const;
export type GetAllAgenciesQueryVariables = Types.Exact<{ [key: string]: never }>;

export type GetAllAgenciesQuery = {
  __typename?: "Query";
  getAllAgencies: Array<{
    __typename?: "Agency";
    id: number;
    name: string;
    address: string;
    city: string;
    country: string;
    email: string;
    phone: string;
    postcode: string;
    productCodes: Array<{
      __typename?: "ProductCode";
      id: number;
      isSizeable: boolean;
      size?: string | null;
      status: Types.Status;
    }>;
    bookings: Array<{ __typename?: "Booking"; id: number; status: Types.StatusBooking; startDate: any; endDate: any }>;
  }>;
};

export const GetAllAgenciesDocument = gql`
  query GetAllAgencies {
    getAllAgencies {
      id
      name
      address
      city
      country
      email
      phone
      postcode
      productCodes {
        id
        isSizeable
        size
        status
      }
      bookings {
        id
        status
        startDate
        endDate
      }
    }
  }
`;

/**
 * __useGetAllAgenciesQuery__
 *
 * To run a query within a React component, call `useGetAllAgenciesQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetAllAgenciesQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetAllAgenciesQuery({
 *   variables: {
 *   },
 * });
 */
export function useGetAllAgenciesQuery(
  baseOptions?: Apollo.QueryHookOptions<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>,
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useQuery<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>(GetAllAgenciesDocument, options);
}
export function useGetAllAgenciesLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>,
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useLazyQuery<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>(GetAllAgenciesDocument, options);
}
export function useGetAllAgenciesSuspenseQuery(
  baseOptions?: Apollo.SuspenseQueryHookOptions<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>,
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useSuspenseQuery<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>(GetAllAgenciesDocument, options);
}
export type GetAllAgenciesQueryHookResult = ReturnType<typeof useGetAllAgenciesQuery>;
export type GetAllAgenciesLazyQueryHookResult = ReturnType<typeof useGetAllAgenciesLazyQuery>;
export type GetAllAgenciesSuspenseQueryHookResult = ReturnType<typeof useGetAllAgenciesSuspenseQuery>;
export type GetAllAgenciesQueryResult = Apollo.QueryResult<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>;
