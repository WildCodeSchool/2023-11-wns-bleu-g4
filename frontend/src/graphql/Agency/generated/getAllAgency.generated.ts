import * as Types from "../../generated/schema";

import * as Apollo from "@apollo/client";
import { gql } from "@apollo/client";
const defaultOptions = {} as const;
export type GetAllAgenciesQueryVariables = Types.Exact<{ [key: string]: never }>;

export type GetAllAgenciesQuery = {
  __typename?: "Query";
  getAllAgencies: Array<{
    __typename?: "Agency";
    postcode: string;
    phone: string;
    name: string;
    id: number;
    email: string;
    country: string;
    city: string;
    address: string;
    productCodes: Array<{
      __typename?: "Product_code";
      size?: string | null;
      isSizeable: boolean;
      status: Types.Status;
      id: number;
    }>;
  }>;
};

export const GetAllAgenciesDocument = gql`
  query GetAllAgencies {
    getAllAgencies {
      postcode
      phone
      name
      id
      email
      country
      city
      address
      productCodes {
        size
        isSizeable
        status
        id
      }
    }
  }
`;

/**
 * __useGetAllAgenciesQuery__
 *
 * To run a query within a React component, call `useGetAllAgenciesQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetAllAgenciesQuery` returns an object from Apollo Client that contains loading,
 *  error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on:
 *  https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetAllAgenciesQuery({
 *   variables: {
 *   },
 * });
 */
export function useGetAllAgenciesQuery(
  baseOptions?: Apollo.QueryHookOptions<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>,
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useQuery<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>(GetAllAgenciesDocument, options);
}
export function useGetAllAgenciesLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>,
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useLazyQuery<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>(GetAllAgenciesDocument, options);
}
export type GetAllAgenciesQueryHookResult = ReturnType<typeof useGetAllAgenciesQuery>;
export type GetAllAgenciesLazyQueryHookResult = ReturnType<typeof useGetAllAgenciesLazyQuery>;
export type GetAllAgenciesQueryResult = Apollo.QueryResult<GetAllAgenciesQuery, GetAllAgenciesQueryVariables>;
