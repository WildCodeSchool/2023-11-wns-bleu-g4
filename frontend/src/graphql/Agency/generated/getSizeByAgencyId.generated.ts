import * as Types from '../../generated/schema';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type GetSizeByAgencyIdQueryVariables = Types.Exact<{
  agencyId: Types.Scalars['Int']['input'];
}>;


export type GetSizeByAgencyIdQuery = { __typename?: 'Query', getAgencyById: { __typename?: 'Agency', productCodes: Array<{ __typename?: 'ProductCode', isSizeable: boolean, size?: string | null, product?: { __typename?: 'Product', id: number } | null }> } };


export const GetSizeByAgencyIdDocument = gql`
    query getSizeByAgencyId($agencyId: Int!) {
  getAgencyById(agencyId: $agencyId) {
    productCodes {
      isSizeable
      size
      product {
        id
      }
    }
  }
}
    `;

/**
 * __useGetSizeByAgencyIdQuery__
 *
 * To run a query within a React component, call `useGetSizeByAgencyIdQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetSizeByAgencyIdQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetSizeByAgencyIdQuery({
 *   variables: {
 *      agencyId: // value for 'agencyId'
 *   },
 * });
 */
export function useGetSizeByAgencyIdQuery(baseOptions: Apollo.QueryHookOptions<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables> & ({ variables: GetSizeByAgencyIdQueryVariables; skip?: boolean; } | { skip: boolean; }) ) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables>(GetSizeByAgencyIdDocument, options);
      }
export function useGetSizeByAgencyIdLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables>(GetSizeByAgencyIdDocument, options);
        }
export function useGetSizeByAgencyIdSuspenseQuery(baseOptions?: Apollo.SuspenseQueryHookOptions<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useSuspenseQuery<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables>(GetSizeByAgencyIdDocument, options);
        }
export type GetSizeByAgencyIdQueryHookResult = ReturnType<typeof useGetSizeByAgencyIdQuery>;
export type GetSizeByAgencyIdLazyQueryHookResult = ReturnType<typeof useGetSizeByAgencyIdLazyQuery>;
export type GetSizeByAgencyIdSuspenseQueryHookResult = ReturnType<typeof useGetSizeByAgencyIdSuspenseQuery>;
export type GetSizeByAgencyIdQueryResult = Apollo.QueryResult<GetSizeByAgencyIdQuery, GetSizeByAgencyIdQueryVariables>;