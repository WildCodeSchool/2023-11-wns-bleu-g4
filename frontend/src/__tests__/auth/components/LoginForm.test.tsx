import "@testing-library/jest-dom";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import LoginForm from "@/features/auth/login/LoginForm";
import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { toast } from 'react-toastify';
import { LoginDocument } from "@/graphql/User/generated/Login.generated";
import { GraphQLError } from "graphql";

jest.mock('react-toastify', () => ({
    toast: {
        success: jest.fn(),
        error  : jest.fn(),
    },
}));

jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: (key: string) => key,
    }),
}));


describe("LoginForm", () => {
    it("should display a Login Form component", () => {
        const loginForm = render(
            <MockedProvider mocks = {[]} addTypename = {false}>
                <LoginForm />
            </MockedProvider>
        );
        expect(loginForm).toMatchSnapshot();
    });

    it("should show error toast when clicking login button with invalid credentials", async () => {
        const mocks: MockedResponse[] = [
            {
                request: {
                    query    : LoginDocument,
                    variables: { "data": { "email": "user@example.com", "password": "password123" } }
                },
                result: {
                    data   : null,
                    errors : [new GraphQLError("INVALID CREDENTIALS")],
                    context: undefined
                },
            },
        ];

        render(
            <MockedProvider mocks = {mocks} addTypename = {false}>
                <LoginForm />
            </MockedProvider>
        );

        const emailInput    = screen.getByPlaceholderText(/Email/i);
        const passwordInput = screen.getByPlaceholderText(/Password/i);
        const button        = screen.getByRole("button", { name: /Login/i });

        fireEvent.change(emailInput, { target: { value: 'user@example.com' } });
        fireEvent.change(passwordInput, { target: { value: 'password123' } });
        fireEvent.click(button);

        await waitFor(() => {
            expect(toast.error).toHaveBeenCalledWith(
                expect.stringContaining("INVALID CREDENTIALS"),
                expect.objectContaining({
                    autoClose      : 2000,
                    closeOnClick   : true,
                    draggable      : true,
                    hideProgressBar: false,
                    pauseOnHover   : true,
                    position       : "bottom-right",
                    theme          : "colored"
                })
            );
        });
    });
});