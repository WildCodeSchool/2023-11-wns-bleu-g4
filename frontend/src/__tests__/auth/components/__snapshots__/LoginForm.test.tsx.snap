// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should display a Login Form component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-card css-1l493p5"
      >
        <div
          class="chakra-card__header css-xi606m"
        >
          <h1
            class="chakra-heading css-12cltki"
          >
            LOGIN
          </h1>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-1xjlnrc"
        />
        <form>
          <div
            class="chakra-card__body css-0"
          >
            <div
              class="css-1q94rgb"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="field-:r0:"
                  id="field-:r0:-label"
                >
                  Email
                </label>
                <div
                  class="chakra-input__group css-bx0blc"
                  data-group="true"
                  data-theme="light"
                >
                  <input
                    aria-required="true"
                    class="chakra-input css-qkm5ls"
                    data-theme="light"
                    id="field-:r0:"
                    name="email"
                    placeholder="Email"
                    required=""
                    type="email"
                  />
                </div>
              </div>
              <div
                class="css-0"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="field-:r1:"
                    id="field-:r1:-label"
                  >
                    Password
                  </label>
                  <div
                    class="chakra-input__group css-bx0blc"
                    data-group="true"
                    data-theme="light"
                  >
                    <input
                      aria-required="true"
                      class="chakra-input css-qkm5ls"
                      data-theme="light"
                      id="field-:r1:"
                      name="password"
                      placeholder="Password"
                      required=""
                      type="password"
                    />
                    <div
                      class="chakra-input__right-element css-vcnsx7"
                      data-theme="light"
                    >
                      <button
                        class="chakra-button css-xsdlqf"
                        data-theme="light"
                        type="button"
                      >
                        Show
                      </button>
                    </div>
                  </div>
                </div>
                <p
                  class="chakra-text  text-center text-sm py-2 css-0"
                >
                  <a
                    class="hover:underline hover:text-orange-500"
                    href="#"
                  >
                    Forgot your password ?
                  </a>
                </p>
              </div>
            </div>
          </div>
          <div
            class="chakra-card__footer css-k008qs"
          >
            <div
              class="w-full css-j7qwjs"
            >
              <button
                class="chakra-button w-full css-2g5nhs"
                type="submit"
              >
                Login
              </button>
              <p
                class="chakra-text  text-center text-sm py-2 css-1xu3tth"
              >
                Not yet registered ?
                 
                <a
                  class="underline text-orange-500"
                  href="/signup"
                >
                  sign up
                </a>
              </p>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-card css-1l493p5"
    >
      <div
        class="chakra-card__header css-xi606m"
      >
        <h1
          class="chakra-heading css-12cltki"
        >
          LOGIN
        </h1>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1xjlnrc"
      />
      <form>
        <div
          class="chakra-card__body css-0"
        >
          <div
            class="css-1q94rgb"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="field-:r0:"
                id="field-:r0:-label"
              >
                Email
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
                data-theme="light"
              >
                <input
                  aria-required="true"
                  class="chakra-input css-qkm5ls"
                  data-theme="light"
                  id="field-:r0:"
                  name="email"
                  placeholder="Email"
                  required=""
                  type="email"
                />
              </div>
            </div>
            <div
              class="css-0"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="field-:r1:"
                  id="field-:r1:-label"
                >
                  Password
                </label>
                <div
                  class="chakra-input__group css-bx0blc"
                  data-group="true"
                  data-theme="light"
                >
                  <input
                    aria-required="true"
                    class="chakra-input css-qkm5ls"
                    data-theme="light"
                    id="field-:r1:"
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                  />
                  <div
                    class="chakra-input__right-element css-vcnsx7"
                    data-theme="light"
                  >
                    <button
                      class="chakra-button css-xsdlqf"
                      data-theme="light"
                      type="button"
                    >
                      Show
                    </button>
                  </div>
                </div>
              </div>
              <p
                class="chakra-text  text-center text-sm py-2 css-0"
              >
                <a
                  class="hover:underline hover:text-orange-500"
                  href="#"
                >
                  Forgot your password ?
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          class="chakra-card__footer css-k008qs"
        >
          <div
            class="w-full css-j7qwjs"
          >
            <button
              class="chakra-button w-full css-2g5nhs"
              type="submit"
            >
              Login
            </button>
            <p
              class="chakra-text  text-center text-sm py-2 css-1xu3tth"
            >
              Not yet registered ?
               
              <a
                class="underline text-orange-500"
                href="/signup"
              >
                sign up
              </a>
            </p>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
